src/model/OutputTableSummaryReports_Impl.hpp

```diff
@@ -0,0 +1,111 @@
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#ifndef MODEL_OUTPUTTABLESUMMARYREPORTS_IMPL_HPP
+#define MODEL_OUTPUTTABLESUMMARYREPORTS_IMPL_HPP
+
+#include <vector>
+#include "ModelAPI.hpp"
+#include "ModelObject_Impl.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  /** OutputTableSummaryReports_Impl is a ModelObject_Impl that is the implementation class for OutputTableSummaryReports.*/
+  class MODEL_API OutputTableSummaryReports_Impl : public ModelObject_Impl {
+  
+  public:
+
+    /** @name Constructors and Destructors */
+    //@{
+
+    OutputTableSummaryReports_Impl(const IdfObject& idfObject,
+                                   Model_Impl* model,
+                                   bool keepHandle);
+
+    OutputTableSummaryReports_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                   Model_Impl* model,
+                                   bool keepHandle);
+
+    OutputTableSummaryReports_Impl(const OutputTableSummaryReports_Impl& other,
+                                   Model_Impl* model,
+                                   bool keepHandle);
+
+    virtual ~OutputTableSummaryReports_Impl() {}
+
+    //@}
+    /** @name Virtual Methods */
+    //@{
+
+    virtual const std::vector<std::string>& outputVariableNames() const override;
+
+    virtual IddObjectType iddObjectType() const override;
+
+    //@}
+    /** @name Getters */
+    //@{
+
+    bool removeSummaryReport(unsigned groupIndex);
+
+    void removeAllSummaryReports();
+
+    std::vector<std::string> summaryReports() const;
+
+    unsigned int numberofSummaryReports() const;
+
+    boost::optional<unsigned> summaryReportIndex(std::string summaryReport) const;
+
+    boost::optional<std::string> getSummaryReport(unsigned groupIndex) const;
+
+    //@}
+    /** @name Setters */
+    //@{
+
+    bool addSummaryReport(std::string summaryReport);
+
+    bool addSummaryReports(const std::vector<std::string> summaryReports);
```

Comment:
By const-reference: `const std::string&`
 and `const std::vector<std::string>& summaryReports`